### PR TITLE
8314544: Matrix multiply benchmark using Vector API

### PR DESCRIPTION
**Pull request by Styp**
Added a bunch of different implementations for Vector API Matrix Multiplications:

- Baseline
- Blocked (Cache Local)
- FMA
- Vector API Simple Implementation
- Vector API Blocked Implementation

Commit was discussed with @PaulSandoz

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314544](https://bugs.openjdk.org/browse/JDK-8314544): Matrix multiply benchmark using Vector API (**Bug** - P4)

### Reviewers
 * [Paul Sandoz](https://openjdk.org/census#psandoz) (@PaulSandoz - **Reviewer**)
 * [Eric Liu](https://openjdk.org/census#eliu) (@e1iu - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15338/head:pull/15338` \
`$ git checkout pull/15338`

Update a local copy of the PR: \
`$ git checkout pull/15338` \
`$ git pull https://git.openjdk.org/jdk.git pull/15338/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15338`

View PR using the GUI difftool: \
`$ git pr show -t 15338`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15338.diff">https://git.openjdk.org/jdk/pull/15338.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15338#issuecomment-1683319155)

### CODE REVIEW

File: test/micro/org/openjdk/bench/jdk/incubator/vector/MatrixMultiplicationBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                        sum.intoArray(result, i * n + j);
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    private static float[] newFloatRowMatrix(int size) {
+        Random rand = new Random();
```
Comment (PaulSandoz) [inline test/micro/org/openjdk/bench/jdk/incubator/vector/MatrixMultiplicationBenchmark.java:176]: Instead we can use the new random generator API e.g,:
```
var rand = RandomGenerator.getDefault();
```
This is not super important, but its good practice to try and use more preferred APIs.

File: test/micro/org/openjdk/bench/jdk/incubator/vector/MatrixMultiplicationBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+package org.openjdk.bench.jdk.incubator.vector;
+
+import java.util.Random;
+
+import jdk.incubator.vector.*;
+
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
```
Comment (PaulSandoz) [inline test/micro/org/openjdk/bench/jdk/incubator/vector/MatrixMultiplicationBenchmark.java:33]: Unused import.

**Comment by Styp** [comment]
Changed according to comments.

**Review by PaulSandoz — APPROVED** [review]

**Review by e1iu — APPROVED** [review]
Looks good to me.

**Comment by sviswa7** [comment]
@Styp Please correct the title in the PR. There is title mismatch between PR and JBS.
PR title: Matrix multiple benchmark using Vector API
"multiple" vs "multiply"

**Comment by Styp** [comment]
fixed typo.

**Comment by sviswa7** [comment]
@Styp Thanks, the next step would be for you to flag this PR as ready for integration by typing /integrate in a new github comment and submit. (Afterwards, your sponsor types /sponsor in a new comment to perform the integration).

**Pull request closed without merging**